pkg/operator/controllers/storageaccounts/storageaccounts.go

```diff
@@ -32,6 +33,10 @@ func (r *reconcileManager) reconcileAccounts(ctx context.Context) error {
 	for _, subnet := range subnets {
 		mgmtSubnet, err := r.subnets.Get(ctx, subnet.ResourceID)
 		if err != nil {
+			if azureerrors.IsNotFoundError(err) {
+				r.log.Infof("Subnet %s not found, skipping", subnet.ResourceID)
+				break
+			}
```

Comment:
while we're here, should we add `429` handling, or are we waiting for SDK Track 2?

Reply:
We perhaps ought to add 429 handling, either before we update to Track 2, or even after (if we want our controllers to requeue themselves when throttled rather than running and waiting for eventual completion), but I think it can and should be done separately. I've included the change in [this PR](https://github.com/Azure/ARO-RP/pull/3272) to the StorageAccounts controller, and we can bring the same refactor to the Subnets controller, or refactor both controllers to perform their work within a single shared Reconcile and share whatever ARM API GET Resource calls they are able to. 

Handling Retry-After is good for dealing with the operator hitting API quota limits - the change in this PR is trying to prevent us from even getting to that state in the first place.